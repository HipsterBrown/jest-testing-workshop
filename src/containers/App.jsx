import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

const App = ({result, params}) => {
  const {component: Component} = result;
  return <Component {...params} />;
};

App.defaultProps = {
  result: null,
};

App.propTypes = {
  params: PropTypes.shape({}),
  result: PropTypes.shape({
    component: PropTypes.func,
  }),
};

export default connect(state => state.router)(App);
export {App as MockApp};
