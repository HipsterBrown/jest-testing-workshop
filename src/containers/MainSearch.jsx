import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {bindActionCreators} from 'redux';
import {search} from '../actions/search';
import NavBar from '../components/NavBar';
import MovieCard from '../components/MovieCard';
import Loading from '../components/Loading';

const MainSearch = props => (
  <div className="sans-serif">
    <NavBar />
    <div className="pa4 center w-60">
      <div className="tc mb5">
        <h1 className="f-headline mt0 mb0">
          Welcome to <br /> Movie Search
        </h1>
        <p className="f3">
          The world's premiere movie search tool built with React and Redux!
        </p>
      </div>

      <input
        className="db center w-100 ph2 pv3 f4 lh-solid"
        type="text"
        placeholder="Wall-E"
        onChange={({target: {value}}) => props.search(value)}
      />

      {props.loading && (
        <div className="tc pa4">
          <Loading fill="#D5008F" height={200} width={200} />
        </div>
      )}

      {!props.loading && (
        <ul className="list ph0 flex flex-wrap">
          {props.results.map(({artworkUrl100, trackName, trackId}) => (
            <li className="w-20 pa2">
              <MovieCard
                link={`/movie/${trackId}`}
                poster={artworkUrl100.replace('100x100', '227x227')}
                title={trackName}
              />
            </li>
          ))}
        </ul>
      )}
    </div>
  </div>
);

MainSearch.defaultProps = {
  error: null,
  loading: false,
  resultCount: 0,
  results: [],
};

MainSearch.propTypes = {
  error: PropTypes.node,
  loading: PropTypes.bool,
  resultCount: PropTypes.number,
  results: PropTypes.arrayOf(PropTypes.shape({})),
  search: PropTypes.func.isRequired,
  query: PropTypes.shape({}).isRequired,
};

function mapStateToProps({router, search}) {
  return {
    ...router,
    ...search,
  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      search,
    },
    dispatch,
  );
}

export default connect(mapStateToProps, mapDispatchToProps)(MainSearch);
export {MainSearch as MockMainSearch};
