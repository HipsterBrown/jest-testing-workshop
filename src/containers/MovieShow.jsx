import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

class MovieShow extends React.Component {
  render() {
    const { id } = this.props;
    return (
      <div className="sans-serif">
        <h1 className="headline">Fill in Details for Movie {id}</h1>
      </div>
    );
  }
}

MovieShow.defaultProps = {
};

MovieShow.propTypes = {
  id: PropTypes.string.isRequired,
};

export default connect(state => state.movie)(MovieShow);
export { MovieShow as MockMovieShow };
