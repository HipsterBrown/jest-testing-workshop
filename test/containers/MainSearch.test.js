import React from 'react';
import {shallow} from 'enzyme';
import renderer from 'react-test-renderer';
import MockRouterProvider from 'helpers/MockRouterProvider';
import {MockMainSearch} from 'containers/MainSearch';

describe('<MainSearch />', () => {
  const search = jest.fn();
  const props = {
    query: {},
    search,
  };

  test('text input triggers search action', () => {
    const value = 'test value';
    const wrapper = shallow(<MockMainSearch {...props} />);

    wrapper.find('input').simulate('change', {target: {value}});
    expect(search).toHaveBeenCalledWith(value);
  });

  test('laoding snanpshot', () => {
    const newProps = {
      ...props,
      loading: true,
    };

    expect(
      renderer
        .create(
          <MockRouterProvider>
            <MockMainSearch {...newProps} />
          </MockRouterProvider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });

  test('empty search snapshot', () => {
    expect(
      renderer
        .create(
          <MockRouterProvider>
            <MockMainSearch {...props} />
          </MockRouterProvider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });

  test('show search results snapshot', () => {
    const newProps = {
      ...props,
      results: [
        {
          artworkUrl100: 'test-url-100x100.png',
          trackName: 'test one',
          trackId: 13,
        },
        {
          artworkUrl100: 'another-test-url-100x100.png',
          trackName: 'test two',
          trackId: 26,
        },
      ],
    };

    expect(
      renderer
        .create(
          <MockRouterProvider>
            <MockMainSearch {...newProps} />
          </MockRouterProvider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
