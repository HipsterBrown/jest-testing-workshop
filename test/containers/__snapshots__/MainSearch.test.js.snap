// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainSearch /> empty search snapshot 1`] = `
<div
  className="sans-serif"
>
  <nav
    className="bg-near-black ph5 pv3 ttu f5 fw6"
  >
    <a
      className="link underline-hover near-white"
      href="/"
      onClick={[Function]}
      target={undefined}
    >
      Home
    </a>
  </nav>
  <div
    className="pa4 center w-60"
  >
    <div
      className="tc mb5"
    >
      <h1
        className="f-headline mt0 mb0"
      >
        Welcome to 
        <br />
         Movie Search
      </h1>
      <p
        className="f3"
      >
        The world's premiere movie search tool built with React and Redux!
      </p>
    </div>
    <input
      className="db center w-100 ph2 pv3 f4 lh-solid"
      onChange={[Function]}
      placeholder="Wall-E"
      type="text"
    />
    <ul
      className="list ph0 flex flex-wrap"
    />
  </div>
</div>
`;

exports[`<MainSearch /> laoding snanpshot 1`] = `
<div
  className="sans-serif"
>
  <nav
    className="bg-near-black ph5 pv3 ttu f5 fw6"
  >
    <a
      className="link underline-hover near-white"
      href="/"
      onClick={[Function]}
      target={undefined}
    >
      Home
    </a>
  </nav>
  <div
    className="pa4 center w-60"
  >
    <div
      className="tc mb5"
    >
      <h1
        className="f-headline mt0 mb0"
      >
        Welcome to 
        <br />
         Movie Search
      </h1>
      <p
        className="f3"
      >
        The world's premiere movie search tool built with React and Redux!
      </p>
    </div>
    <input
      className="db center w-100 ph2 pv3 f4 lh-solid"
      onChange={[Function]}
      placeholder="Wall-E"
      type="text"
    />
    <div
      className="tc pa4"
    >
      <svg
        fill="#D5008F"
        height={200}
        viewBox="0 0 105 105"
        width={200}
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="12.5"
          cy="12.5"
          r="12.5"
        >
          <animate
            attributeName="fill-opacity"
            begin="0s"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="1;.2;1"
          />
        </circle>
        <circle
          cx="12.5"
          cy="52.5"
          fill-opacity=".5"
          r="12.5"
        >
          <animate
            attributeName="fill-opacity"
            begin="100ms"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="1;.2;1"
          />
        </circle>
        <circle
          cx="52.5"
          cy="12.5"
          r="12.5"
        >
          <animate
            attributeName="fill-opacity"
            begin="300ms"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="1;.2;1"
          />
        </circle>
        <circle
          cx="52.5"
          cy="52.5"
          r="12.5"
        >
          <animate
            attributeName="fill-opacity"
            begin="600ms"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="1;.2;1"
          />
        </circle>
        <circle
          cx="92.5"
          cy="12.5"
          r="12.5"
        >
          <animate
            attributeName="fill-opacity"
            begin="800ms"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="1;.2;1"
          />
        </circle>
        <circle
          cx="92.5"
          cy="52.5"
          r="12.5"
        >
          <animate
            attributeName="fill-opacity"
            begin="400ms"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="1;.2;1"
          />
        </circle>
        <circle
          cx="12.5"
          cy="92.5"
          r="12.5"
        >
          <animate
            attributeName="fill-opacity"
            begin="700ms"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="1;.2;1"
          />
        </circle>
        <circle
          cx="52.5"
          cy="92.5"
          r="12.5"
        >
          <animate
            attributeName="fill-opacity"
            begin="500ms"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="1;.2;1"
          />
        </circle>
        <circle
          cx="92.5"
          cy="92.5"
          r="12.5"
        >
          <animate
            attributeName="fill-opacity"
            begin="200ms"
            calcMode="linear"
            dur="1s"
            repeatCount="indefinite"
            values="1;.2;1"
          />
        </circle>
      </svg>
    </div>
  </div>
</div>
`;

exports[`<MainSearch /> show search results snapshot 1`] = `
<div
  className="sans-serif"
>
  <nav
    className="bg-near-black ph5 pv3 ttu f5 fw6"
  >
    <a
      className="link underline-hover near-white"
      href="/"
      onClick={[Function]}
      target={undefined}
    >
      Home
    </a>
  </nav>
  <div
    className="pa4 center w-60"
  >
    <div
      className="tc mb5"
    >
      <h1
        className="f-headline mt0 mb0"
      >
        Welcome to 
        <br />
         Movie Search
      </h1>
      <p
        className="f3"
      >
        The world's premiere movie search tool built with React and Redux!
      </p>
    </div>
    <input
      className="db center w-100 ph2 pv3 f4 lh-solid"
      onChange={[Function]}
      placeholder="Wall-E"
      type="text"
    />
    <ul
      className="list ph0 flex flex-wrap"
    >
      <li
        className="w-20 pa2"
      >
        <div
          className="tc"
        >
          <img
            alt="test one"
            className="db w-100 ba b--black-10 mb3"
            src="test-url-227x227.png"
          />
          <a
            className="link underline-hover near-black f5 fw6"
            href="/movie/13"
            onClick={[Function]}
            target={undefined}
          >
            test one
          </a>
        </div>
      </li>
      <li
        className="w-20 pa2"
      >
        <div
          className="tc"
        >
          <img
            alt="test two"
            className="db w-100 ba b--black-10 mb3"
            src="another-test-url-227x227.png"
          />
          <a
            className="link underline-hover near-black f5 fw6"
            href="/movie/26"
            onClick={[Function]}
            target={undefined}
          >
            test two
          </a>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
